File: spring-boot-jersey/src/main/java/com/example/domain/LocalDateTimeConverter.java
package com.example.domain;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;
import java.sql.Timestamp;
import java.time.LocalDateTime;

@Converter(autoApply = true)
public class LocalDateTimeConverter implements AttributeConverter<LocalDateTime, Timestamp> {

    @Override
    public Timestamp convertToDatabaseColumn(LocalDateTime dateTime) {
        return dateTime == null ? null : Timestamp.valueOf(dateTime);

    }

    @Override
    public LocalDateTime convertToEntityAttribute(Timestamp timestamp) {
        return timestamp == null ? null : timestamp.toLocalDateTime();
    }
}

File: spring-boot-scala/src/main/scala/com/example/core/LocalDateTimeConverter.scala
package com.example.core

import java.sql.Timestamp
import java.time.LocalDateTime
import javax.persistence.{AttributeConverter, Converter}

@Converter(autoApply = true)
class LocalDateTimeConverter extends AttributeConverter[LocalDateTime, Timestamp] {

  override def convertToDatabaseColumn(localDateTime: LocalDateTime): Timestamp = Timestamp.valueOf(localDateTime)

  override def convertToEntityAttribute(timestamp: Timestamp): LocalDateTime = timestamp.toLocalDateTime
}
